test/it/group-list.js

```diff
@@ -32,6 +32,6 @@ describe('Group API tests', () => {
     expect(groupPresent).to.equal(true);
 
     // 3. Delete the group
-    await client.deleteGroup(createdGroup.id);
+    await utils.cleanup(client, null, createdGroup);
```

Comment:
Initially when we wrote tests, there was a before() block in which entities were created, then multiple it() blocks that run individual API calls and an after() block that did the cleanup.
But then we decided to move to a single large it() block because individual it() blocks couldn't be run independently in the former case. 
Once we moved to one large it() block, we decided to move the cleanup() inside the it() block itself, since we anyway have only 1 large it() block.
Come to think of it, I still prefer having an after() block that does the cleanup as this block will run even if there is an error in the test. Needs some more discussion I believe...